### PR TITLE
feat(native_transform): add auxiliary transform helpers (PR B8)

### PR DESCRIPTION
**Pull request by sallycr**
## Summary

Ports the `native_transform` "aux island" — `ClipAndScale`, `TimeDuration`, and the imperative `generate_*_transformation` tensor_map helpers — to `lib/native_transform/torch/`. This is PR B8 of the native_transform migration (see `oss_design.md` PR B breakdown).

**Self-contained island, off the assembler-critical path.** `ClipAndScale`/`TimeDuration` are not registered in `TORCH_TRANSFORM_LAYERS_DICT`; `transform_utils.py` composes them (and the already-merged `IDHashTokenizer` wrapper layer from PR B4) imperatively into a `tensor_map`, an alternate usage path distinct from the main `TransformSpec`/`TorchTransformModule` DAG machinery. Verified via grep that nothing else in `native_transform/` imports these three modules, in both this repo and internal.

## Changes

- `lib/native_transform/torch/scale.py` — `ClipAndScale`: clip + scale + optional dtype cast over one or more concatenated input columns.
- `lib/native_transform/torch/duration.py` — `TimeDuration`: floored, unit-scaled duration between two timestamp tensors, with optional log scaling and post-log clipping.
- `lib/native_transform/torch/transform_utils.py` — `generate_numerical_scaled_transformation`, `generate_concatenation_transformation`, `generate_cast_transformation`, `generate_duration_transformation`, `generate_idhash_tokenization_transformation`, `update_output_tensor_map`.
- `lib/native_transform/torch/__init__.py` — export `ClipAndScale` and `TimeDuration` at the package root, alongside the existing layer exports.
- Scrubbed an internal source-attribution URL comment that was present at the top of the internal `scale.py`/`duration.py`/`transform_utils.py` files; behavior is otherwise a faithful port (clipping bounds, duration unit conversion, log-before-clip ordering, empty-vocabulary skip, and other edge cases match internal's test coverage).
- No deliberate behavior changes vs. internal.

## Tests

- New: `tests/test_scale.py`, `tests/test_duration.py`, `tests/test_transform_utils.py` — forward-semantics coverage matching internal's edge cases (clipping, scaling, dtype resolution/errors, 1D unsqueeze, reshaping, log-scale-before-clip ordering, missing-feature no-op, empty-vocabulary skip, custom output columns) plus `torch.jit.script`/`torch.jit.trace` round-trip tests (script+save+load, and trace across dtypes) for the two new layers.
- Full `lib/native_transform/` suite: 326 passed, 1 skipped (pre-existing skip, unrelated to this PR).
- Coverage on the three new modules: 99.19% (100% on `scale.py`/`duration.py`; one partial branch in `transform_utils.py`).
- `ruff format --check` and `ruff check`: clean.

### CODE REVIEW

File: python/michelangelo/lib/native_transform/torch/__init__.py [hunk 1]
```diff
@@ -25,6 +25,8 @@
     Tile,
     TorchTransformBaseLayer,
 )
+from michelangelo.lib.native_transform.torch.duration import TimeDuration
+from michelangelo.lib.native_transform.torch.scale import ClipAndScale
```
Comment (kechen2013) [inline python/michelangelo/lib/native_transform/torch/__init__.py]: Wondering if the `generate_*_transformation` helpers from `transform_utils` should be exported here too? B6 exports `generate_layer_name`/`get_transform_module`, so functions don't seem excluded by convention.

Comment (sallycr) [reply python/michelangelo/lib/native_transform/torch/__init__.py]: Good catch, done in 0679f2b7 -- re-exported all the generate_*_transformation helpers and update_output_tensor_map from the package __init__, matching B6's convention of exporting functions alongside classes.

**Review by kechen2013 — APPROVED** [review]
